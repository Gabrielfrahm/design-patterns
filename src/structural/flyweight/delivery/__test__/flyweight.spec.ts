import { deliveryContext } from "../delivery-context";
import { DeliveryFactory } from "../delivery-factory";

describe("flyweight test", () => {
  it("factory", () => {
    const factory = new DeliveryFactory();
    deliveryContext(factory, "client 1", "12", "av dale", "sp");
    deliveryContext(factory, "client 2", "44", "av salame", "sp");
    deliveryContext(factory, "client 3", "455535", "av dale", "pinda");
    deliveryContext(factory, "client 4", "546", "av dale", "sp");

    expect(factory.getLocations()).toMatchSnapshot({
      avdale_pinda: { intrinsicState: { city: "pinda", street: "av dale" } },
      avdale_sp: { intrinsicState: { city: "sp", street: "av dale" } },
      avsalame_sp: { intrinsicState: { city: "sp", street: "av salame" } },
    });
  });
});
