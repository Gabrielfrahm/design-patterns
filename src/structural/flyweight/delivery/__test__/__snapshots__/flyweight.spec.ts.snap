// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flyweight test factory 1`] = `
{
  "avdale_pinda": {
    "intrinsicState": {
      "city": "pinda",
      "street": "av dale",
    },
  },
  "avdale_sp": {
    "intrinsicState": {
      "city": "sp",
      "street": "av dale",
    },
  },
  "avsalame_sp": {
    "intrinsicState": {
      "city": "sp",
      "street": "av salame",
    },
  },
}
`;
